### ISSUE TITLE
Flow ID allocation failures

### ISSUE THREAD

**Issue by LPardue** [ietf-wg-masque/draft-ietf-masque-h3-datagram#30]
draft 00 section 6 says:

```
   If the flow identifier allocation service
   of an endpoint runs out of values lower than 10^15-1, the endpoint
   MUST fail the flow identifier allocation.
```

First, Section 3, which talks about the service, seems like a better place to more-generally talk about errors. Specific error conditions probably depend on the encoding - in this case, the maximum depends on Datagram-Flow-Id header but we might imagine other uses of H3 DGRAM might have other limits.

Second, what does failure look like? Is it a "won't vend flow ID" or is this a connection-critical error etc.

**Comment by DavidSchinazi** [comment]
From the flow allocation service's perspective, it just reports the failure to the requesting application context. Applications need to handle this and decide how. For example, a CONNECT-UDP client should handle failure by opening a second connection, but keeping the current one open until all its streams are done.

**Comment by LPardue** [comment]
that make sense. But there is also an implication that the flow ID allocation service needs to be primed with some knowledge of the bounds for which it can allocate, which might be application-specific. So it makes me wonder what the benefit of putting it in the allocation service itself is. An alternative is that the allocation service provides any legal wire-format value, and the application could enforce the 10^15-1 limit.

**Comment by DavidSchinazi** [comment]
That's a fair point. In practice I don't expect this to ever happen (a given QUIC connection can only send 2^62 packets after all, and that number goes down to 2^23 if using AES), so it might be best to go for the simpler option and just declare 10^15 as the max in the allocation service...

**Comment by LPardue** [comment]
Introducing a hard 10^15  limit would also go some way to addressing https://github.com/ietf-wg-masque/draft-ietf-masque-h3-datagram/issues/31

**Comment by DavidSchinazi** [comment]
This issue is no longer relevant now that we have merged #52, closing.
